src/cli/miner_cli_genesis.erl

```diff
@@ -323,6 +326,136 @@ genesis_proof(["genesis", "proof", PrivKeyB58], [], []) ->
+                lists:foldl(
+                    fun({Address, Entry}, Acc) ->
+                        case blockchain_ledger_data_credits_entry_v1:balance(Entry) of
+                            0 -> Acc;
+                            undefined -> Acc;
+                            Balance -> [blockchain_txn_dc_coinbase_v1:new(Address, Balance) | Acc]
+                        end
+                    end, [], DCAccounts),
+
+            %% get all security accounts and generate a blockchain_txn_security_coinbase_v1 for each
+            SecurityAccounts = blockchain_ledger_v1:snapshot_security_accounts(Ledger),
+            NewSecurityAccountTxns =
+                lists:foldl(
+                    fun({Address, Entry}, Acc) ->
+                        case blockchain_ledger_security_entry_v1:balance(Entry) of
+                            0 -> Acc;
+                            undefined -> Acc;
+                            Balance -> [blockchain_txn_security_coinbase_v1:new(Address, Balance) | Acc]
+                        end
+                    end, [], SecurityAccounts),
+
+            %% iterate over validators and generate a gen txn for each
+            ValFun =
+                fun(V, Acc) ->
+                    ValAddr = blockchain_ledger_validator_v1:address(V),
+                    ValOwner = blockchain_ledger_validator_v1:owner_address(V),
+                    ValStake = blockchain_ledger_validator_v1:stake(V),
+                    [blockchain_txn_gen_validator_v1:new(ValAddr, ValOwner, ValStake) | Acc]
+                end,
+            NewGenValTxns = blockchain_ledger_v1:fold_validators(ValFun, [], Ledger),
+
+            %% get all active gateways and generate a gen txn for each
+            %% GWs without a location are excluded
+            GWs = blockchain_ledger_v1:snapshot_gateways(Ledger),
+            {NewGenGatewayTxns, ValAddrs} =
+                lists:foldl(
+                    fun({GWAddr, GW}, {TxnAcc, AddrAccc} = Acc) ->
+                        GWOwnerAddr = blockchain_ledger_gateway_v2:owner_address(GW),
+                        case blockchain_ledger_gateway_v2:location(GW) of
+                            [] -> Acc;
+                            undefined -> Acc;
+                            GWLoc ->
+                                GWNonce = blockchain_ledger_gateway_v2:nonce(GW),
+                                {[blockchain_txn_gen_gateway_v1:new(GWAddr, GWOwnerAddr, GWLoc, GWNonce) | TxnAcc],
+                                    [GWAddr | AddrAccc]}
+                        end
+                    end, {[], []}, GWs),
+
+            %% create a consensus group txn
+            {ok, N} = blockchain:config(?num_consensus_members, Ledger),
+            Addrs = [Addr || Addr <- ValAddrs ],
+            SortedAddrs = lists:sort(Addrs),
+            ConsensusAddrs = lists:sublist(SortedAddrs, 1, N),
+            NewConsensusGroupTxn = blockchain_txn_consensus_group_v1:new(ConsensusAddrs, <<>>, 1, 0),
```

Comment:
The risk here is that you pick some dead consensus members, how do you boot a chain with this?

Reply:
i could check for liveness based on last heartbeat, exclude any which are not recent from the sorted list

Reply:
Another option would be to carry across the existing CG members to the new group

Reply:
I think it'd be better to not include a consensus group txn, and let initial_dkg handle it?

Reply:
Then you could simply boot the chain on the new validators by giving them the exported genesis txns and the initial consensus group members.

Reply:
https://github.com/helium/miner/blob/master/src/miner_consensus_mgr.erl#L947-L958

So really you're not making the genesis block directly, you're making the contents of the genesis block